plugins/modules/helm_cli.py

```diff
@@ -0,0 +1,593 @@
+                release = release_lower
+                if release['name'] == release_name and release['namespace'] == release_namespace:
+                    return release
+        else:
+            for release in state:
+                if release['name'] == release_name and release['namespace'] == release_namespace:
+                    return release
+    return None
+
+
+# Get Release state from deployed release
+def get_release_status(command, release_name, release_namespace):
+    list_command = command + " list --output=yaml "
+
+    if not is_helm_2:
+        list_command += " --namespace=" + release_namespace
+        list_command += " --filter "
+
+    list_command += release_name
+
+    rc, out, err = module.run_command(list_command)
+
+    if rc != 0:
+        module.fail_json(
+            msg="Failure when executing Helm command. Exited {0}.\nstdout: {1}\nstderr: {2}".format(rc, out, err),
+            command=list_command
+        )
+
+    release = get_release(yaml.safe_load(out), release_name, release_namespace)
+
+    if release is None:  # not install
+        return None
+
+    release['values'] = get_values(command, release_name, release_namespace)
+
+    return release
+
+
+# Run Repo update
+def run_repo_update(command):
+    repo_update_command = command + " repo update"
+
+    rc, out, err = module.run_command(repo_update_command)
+    if rc != 0:
+        module.fail_json(
+            msg="Failure when executing Helm command. Exited {0}.\nstdout: {1}\nstderr: {2}".format(rc, out, err),
+            command=repo_update_command
+        )
+
+
+# Get chart info
+def fetch_chart_info(command, chart_ref):
+    if is_helm_2:
+        inspect_command = command + " inspect chart " + chart_ref
+    else:
+        inspect_command = command + " show chart " + chart_ref
+
+    rc, out, err = module.run_command(inspect_command)
+    if rc != 0:
+        module.fail_json(
+            msg="Failure when executing Helm command. Exited {0}.\nstdout: {1}\nstderr: {2}".format(rc, out, err),
+            command=inspect_command
+        )
+
+    return yaml.safe_load(out)
+
+
+# Install/upgrade/rollback release chart
+def deploy(command, release_name, release_namespace, release_values, chart_name, wait, wait_timeout, disable_hook, force):
+    deploy_command = command + " upgrade -i"  # install/upgrade
+
+    # Always reset values to keep release_values equal to values released
+    deploy_command += " --reset-values"
+
+    if wait:
+        deploy_command += " --wait"
+        if wait_timeout is not None:
+            deploy_command += " --timeout " + wait_timeout
+
+    if force:
+        deploy_command += " --force"
+
+    if disable_hook:
+        deploy_command += " --no-hooks"
+
+    if release_values != {}:
+        try:
+            import tempfile
+        except ImportError:
+            module.fail_json(msg="Could not import the tempfile python module. Please install `tempfile` package.")
```

Comment:
Could you please use `missing_required_lib` from basic.py?

Reply:
Fixed in last commit